.github/workflows/ci-release.yaml

```diff
@@ -0,0 +1,60 @@
+name: Release CI
+on:
+  push:
+    tags:
+      - 'release/v?[0-9]*'
+
+env:
+  RELEASE_TAG: ${{ github.ref_name }}
+
+jobs:
+  publish:
+    name: Publish to Packagist
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          path: main
+      - name: Set up environment variables
+        run: ./.github/scripts/publish_env.sh >> $GITHUB_ENV
+        working-directory: ./main
```

Comment:
**Question**: Is there a reason why we checkout the code into a subdirectory called `./main`?

I'm presuming it's for consistency (and ease of debugging) with the fact that we stick packed output into `./dist` (and for some API clients we also have a `./temp`).

Reply:
yes. that's why.
to be specific, the Perl client will pack everything under the API client project root, even temp files.  So I need to put the temporary files branching off the parent level of the project root to prevent them from being added.